DefaultConfigsExtension/src/test/java/de/viadee/xai/anchor/adapter/tabular/discretizer/PercentileMedianDiscretizerTest.java

```diff
@@ -170,4 +168,82 @@ public void testUnApply() {
             disc.unApply(13);
         });
     }
+
+    @Test
+    public void testFitNonNumeric() {
+        PercentileMedianDiscretizer disc = new PercentileMedianDiscretizer(2, -1);
+        String[] noNumbers = {"0", "0", "0", "1", "2", "3"};
+
+        assertThrows(ClassCastException.class, () -> {
+            disc.fit(noNumbers);
+        });
+    }
+
+    @Test
+    public void testFitSmallNrOfValues() {
+        PercentileMedianDiscretizer disc = new PercentileMedianDiscretizer(10, -1);
+        Number[] numbers = {0, 1, 2, 3};
+        disc.fit(numbers);
+        try {
+            Field discretizerRelationsField = PercentileMedianDiscretizer.class.getDeclaredField("discretizerRelations");
+            discretizerRelationsField.setAccessible(true);
+            ArrayList<DiscretizerRelation> scvrList = (ArrayList<DiscretizerRelation>) discretizerRelationsField.get(disc);
+
+            assertEquals(4, scvrList.size(), "tests if number of Relations is set correctly if Values > classCount");
+        } catch (Exception e) {
```

Comment:
There is no need to catch and print exceptions in tests - we would see any exception thrown in the log anyway.